### ISSUE TITLE
Asking for at least one thing

### ISSUE THREAD

**Issue by joelanman** [alphagov/govuk-design-system-backlog#214]
## What

Sometimes you need to ask the user for at least one thing - most commonly at least one form of contact (email, phone number)

For example:

![image](https://user-images.githubusercontent.com/1132904/82566632-1a6bb780-9b74-11ea-919e-22a0f0e49277.png)

![image](https://user-images.githubusercontent.com/1132904/82566679-30797800-9b74-11ea-876b-dacd7b5f1748.png)

## Why

This is used in Register to vote

**Comment by terrysimpson99** [comment]
This follows a slack discussion. We replaced a conditional radio button design (similar to the checkbox design above) with the target fields. We implemented: 
![image](https://user-images.githubusercontent.com/48089049/82568838-4dfc1100-9b77-11ea-8911-3e4d6f0d686e.png)

The phone number and email fields are neither wholly mandatory nor wholly optional. We decided saying 'Optional' was not appropriate. We decided not to explain the validation rule in advance but it's something we could consider.

**Comment by adamsilver** [comment]
I like the checkboxes option @joelanman but as an alternative pattern where the user has to fill out at least one field you could put those fields in a group like this:

![image](https://user-images.githubusercontent.com/37163/82815686-8cecd800-9e91-11ea-8c71-7eb177a2dfef.png)

**Comment by MalcolmVonMoJ** [comment]
We have a similar situation in one of the services here at the LAA.  

Not including the above approaches, we have the following options.  

**1, Not repeat the error next to the question.**
Problem: forces users to look at the top of the page for the error
![image](https://user-images.githubusercontent.com/32877315/83531201-90610e80-a4e4-11ea-9205-157c47b37d07.png)

**2, Repeat the error next to both questions.**
Problem: it refers to both Q1 and Q2 next to each question.  This is similar to Terry Simpson's approach above, but the error summary doesn't duplicate the error.
![image](https://user-images.githubusercontent.com/32877315/83531637-28f78e80-a4e5-11ea-94d6-834df076fd7c.png)

**3, Only repeat the error once, next to the first question.**
Problem: Q2 doesn't have an error, although it is highlighted.  We are relying on the user to look at the error next to Q1.  There are probably accessibility issues here too.  This differs from Adam's approach above in that we have two distinct questions, rather than one question with multiple answers.  
![image](https://user-images.githubusercontent.com/32877315/83531540-06fe0c00-a4e5-11ea-93cd-347e56e490f5.png)

**4, Individual text for each box.**
Problem: It might imply that both are mandatory, which isn't the case.
![image](https://user-images.githubusercontent.com/32877315/83531811-59d7c380-a4e5-11ea-88de-da9eb1e62fd5.png)

**5, Enter something more bespoke which might be more helpful.**
![image](https://user-images.githubusercontent.com/32877315/83532234-dec2dd00-a4e5-11ea-831b-da04ee38ec66.png)

The other thing we are considering is how to word the error summary.  The prevailing view is currently split the single error in two, linking to their respective questions, like this:
![image](https://user-images.githubusercontent.com/32877315/83532325-00bc5f80-a4e6-11ea-9114-6d05ca0bc482.png)

Discussions continue...

**Comment by joelanman** [comment]
@MalcolmVonMoJ what's the wider context to these two questions? Is there more content above them, or a previous page?

**Comment by MalcolmVonMoJ** [comment]
There is no previous page.  Here is a screenshot of the whole question.
![image](https://user-images.githubusercontent.com/32877315/83617166-9b697c80-a580-11ea-9fe5-224c78dd3326.png)

**Comment by MalcolmVonMoJ** [comment]
Our latest design thinking is to move the error message above both questions like this:

![image](https://user-images.githubusercontent.com/32877315/84489185-30fbbf00-ac99-11ea-99c5-ffdcae444887.png)

Regarding screen readers, we don't want the error to be read out and then the question, as this might confuse things when someone doesn't have the benefit of the red line.  

So we have made the visible error `aria-hidden:true` whilst repeating the error in the usual places but using the `govuk-visually-hidden` class.  A screen reader user will thus hear things in the tried and tested order.  

Clicking on the error summary links will jump to the error message at the top so it remains visible, whilst focussing on the input box corresponding with the selected link.

**Comment by anevins12** [comment]
Sounds good, but worth considering that hiding the visible text may be unpredictable to people that have vision and are also using a screen reader. By unpredictable I mean that it's just a little weird because the visual experience is different from the hearing experience.

**Comment by MalcolmVonMoJ** [comment]
That is true, but we are only moving two lines around so hopefully anyone who is using the screen reader as an aide, who still has the benefit of the red line, will not be confused.  

The risk of someone relying on the screen reader for all information hearing "enter A or B", followed by "enter A" might think the choice has been made for them.
